apps/service_persist/lib/persist/compaction.ex

```diff
@@ -0,0 +1,38 @@
+defmodule Persist.Compaction do
+  use GenServer, restart: :transient
+  use Properties, otp_app: :service_persist
+  require Logger
+
+  @instance Persist.Application.instance()
+  getter(:compactor, default: Persist.Compactor.Presto)
+
+  @type init_opts :: [
+          load: Load.Persist.t()
+        ]
+
+  @spec start_link(init_opts) :: GenServer.on_start()
+  def start_link(init_opts) do
+    server_opts = Keyword.take(init_opts, [:name])
```

Comment:
I felt it read a little better and now if the user does not pass name we pass an empty list to GenServer.start_link rather than a keyword list of [name: nil].  I can change it back if we feel it is harder to understand.